operators/advanced_network/python/adv_network_common_pybind.cpp

```diff
@@ -17,109 +17,316 @@
-  m.def("print_stats", (&print_stats), "Print statistics from the advanced_network manager");
-  // m.def("set_cpu_udp_payload",
-  //     [](BurstParams *burst, int idx, long int data, int len) {
-  //             return set_cpu_udp_payload(burst, idx,
-  //                     reinterpret_cast<void*>(data), len); },
-  //             "Set UDP header parameters and copy payload");
-  // m.def("set_cpu_udp_payload",
-  //     [](std::shared_ptr<BurstParams> burst, int idx, long int data, int len) {
-  //         return set_cpu_udp_payload(burst, idx,
-  //              reinterpret_cast<void*>(data), len); },
-  //         "Set UDP header parameters and copy payload");
+        "Get TX packet burst",
+        "burst"_a);
+  m.def("send_tx_burst",
+        py::overload_cast<BurstParams*>(&send_tx_burst),
+        "Send a TX burst",
+        "burst"_a);
+
+  // RX burst functions - all overloads
+  m.def(
+      "get_rx_burst",
+      [](int port, int q) {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr, port, q);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst for specific port and queue",
+      "port"_a,
+      "q"_a);
+
+  m.def(
+      "get_rx_burst",
+      [](int port) {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr, port);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst from any queue on specific port",
+      "port"_a);
 
+  m.def(
+      "get_rx_burst",
+      []() {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst from any queue on any port");
+
+  // Header setting functions
+  m.def("set_eth_header",
+        py::overload_cast<BurstParams*, int, char*>(&set_eth_header),
+        "Set Ethernet header in packet",
+        "burst"_a,
+        "idx"_a,
+        "dst_addr"_a);
+  m.def("set_ipv4_header",
+        py::overload_cast<BurstParams*, int, int, uint8_t, unsigned int, unsigned int>(
+            &set_ipv4_header),
+        "Set IPv4 header in packet",
+        "burst"_a,
+        "idx"_a,
+        "ip_len"_a,
+        "proto"_a,
+        "src_host"_a,
+        "dst_host"_a);
+  m.def("set_udp_header",
+        py::overload_cast<BurstParams*, int, int, uint16_t, uint16_t>(&set_udp_header),
+        "Set UDP header in packet",
+        "burst"_a,
+        "idx"_a,
+        "udp_len"_a,
+        "src_port"_a,
+        "dst_port"_a);
+  m.def("set_udp_payload",
+        py::overload_cast<BurstParams*, int, void*, int>(&set_udp_payload),
+        "Set UDP payload in packet",
+        "burst"_a,
+        "idx"_a,
+        "data"_a,
+        "len"_a);
```

Comment:
for void* pointer and size as input, the Python binding probably needs a lambda accepting a Python buffer object like this:

```suggestion
  m.def("set_udp_payload",
      [](BurstParams* burst, int idx, py::buffer data) -> Status {
        py::buffer_info buf_info = data.request();
        return set_udp_payload(burst, idx, buf_info.ptr, buf_info.size);
      },
      "Set UDP payload in packet",
      "burst"_a,
      "idx"_a,
      "data"_a);
```

then Python code could pass a byte string, `bytes` object or other object implementing the buffer protocol (e.g. a NumPy array) as the data.

Reply:
Updated, according to the suggestion

Reply:
Changes made as suggested.